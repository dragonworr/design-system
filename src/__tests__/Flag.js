import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Flag } from '..'

describe('Flag', () => {
  test('renders', () => {
    const json = renderer.create(<Flag />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with width prop', () => {
    const json = renderer.create(<Flag width={256} />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
