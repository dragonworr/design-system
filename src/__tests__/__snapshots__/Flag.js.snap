// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag renders 1`] = `
.c1 {
  margin-bottom: 8px;
  margin-left: -8px;
  margin-right: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: block;
  width: 4px;
  height: 4px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin-right: -8px;
  margin-bottom: -8px;
  border-width: 4px;
  border-style: solid;
  border-color: transparent;
  border-top-color: undefined;
  border-right-color: undefined;
}

.c4 {
  display: inline-block;
  width: 18px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  min-height: 32px;
  margin-left: -10px;
  -webkit-transform: skew(-14deg);
  -ms-transform: skew(-14deg);
  transform: skew(-14deg);
}

.c3 {
  font-size: px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 0 0 0;
  color: white;
  background-color: green;
}

<div
  className="c0 c1"
  width={undefined}
>
  <div
    className="c2"
    color="green"
  />
  <div
    className="c3"
    color="white"
  />
  <div
    className="c4"
    color="green"
  />
</div>
`;

exports[`Flag renders with width prop 1`] = `
.c1 {
  margin-bottom: 8px;
  margin-left: -8px;
  margin-right: 16px;
  width: 256px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: block;
  width: 4px;
  height: 4px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin-right: -8px;
  margin-bottom: -8px;
  border-width: 4px;
  border-style: solid;
  border-color: transparent;
  border-top-color: undefined;
  border-right-color: undefined;
}

.c4 {
  display: inline-block;
  width: 18px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  min-height: 32px;
  margin-left: -10px;
  -webkit-transform: skew(-14deg);
  -ms-transform: skew(-14deg);
  transform: skew(-14deg);
}

.c3 {
  font-size: px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 0 0 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  color: white;
  background-color: green;
}

<div
  className="c0 c1"
  width={256}
>
  <div
    className="c2"
    color="green"
  />
  <div
    className="c3"
    color="white"
  />
  <div
    className="c4"
    color="green"
  />
</div>
`;
